woodwork/utils.py

```diff
@@ -298,3 +299,149 @@ def _parse_logical_type(logical_type, name):
+            raise ValueError('Cannont include None in list of objects to concatenate')
+        elif isinstance(obj.ww.schema, ww.table_schema.TableSchema):
+            drop_cols = []
+            # Keep the first occurance of a key in metadata
+            table_metadata = {**obj.ww.metadata, **table_metadata}
+
+            # Combine table names
+            if obj.ww.name is not None:
+                if table_name:
+                    table_name += '_'
+                table_name += str(obj.ww.name)
+
+            # Handle table indexes
+            for idx_type, obj_idx, table_idx in [('index', obj.ww.index, index),
+                                                 ('time_index', obj.ww.time_index, time_index)]:
+                if obj_idx is not None:
+                    if table_idx is None:
+                        if idx_type == 'index':
+                            index = obj_idx
+                        elif idx_type == 'time_index':
+                            time_index = obj_idx
+                    elif obj_idx != table_idx:
+                        raise IndexError(f'Cannot concat dataframes with different Woodwork {idx_type} columns.')
+                    elif axis in {1, 'columns'} and obj_idx == table_idx:
+                        # Do not include the column from the dataframe to avoid duplicates columns
+                        drop_cols.append(table_idx)
+
+            # To avoid changing input objects, create a new DataFrame without the index columns
+            if drop_cols:
+                obj_to_add = obj.ww.drop(columns=drop_cols)
+            ww_columns = obj_to_add.ww.schema.columns
+        elif isinstance(obj.ww.schema, ww.column_schema.ColumnSchema):
+            ww_columns = {obj.name: obj.ww.schema}
+
+        concat_objs.append(obj_to_add)
+
+        # Compile the typing information per column
+        for name, col_schema in ww_columns.items():
+            # Do not look at typing information for duplicate columns.
+            # Means that index columns will only have first occurrance's typing information
+            if name in col_names_seen:
+                continue
+            logical_types[name] = col_schema.logical_type
+            semantic_tags[name] = col_schema.semantic_tags - {'time_index'} - {'index'}
+            col_metadata[name] = col_schema.metadata
+            col_descriptions[name] = col_schema.description
+            use_standard_tags[name] = col_schema.use_standard_tags
+
+            col_names_seen.add(name)
+
+    # Perform concatenation with the correct library
+    dd = import_or_none('dask.dataframe')
+    ks = import_or_none('databricks.koalas')
+
+    lib = pd
+    if ks and isinstance(obj, (ks.Series, ks.DataFrame)):
+        lib = ks
+    elif dd and isinstance(obj, (dd.Series, dd.DataFrame)):
+        lib = dd
```

Comment:
I think these checks rely on the last value of `obj` that was set in the for loop above, and will check the library of the last value in the list. Might be safer to use `objs[0]` instead of `obj` here, which will of course check the first value instead.

Reply:
updated to use `objs[0]`